### PR TITLE
PCHR-1690: Fix for job role ui updates

### PR DESCRIPTION
**Pull request by bmnepali**
### Issue:
1. Collapse Arrow in Job Role Detail, pointing to wrong direction by default.
2. Not displaying confirmation alert on deleting Job roles.
3. Spinner not displaying until the job roles data is loaded.

### Solution:
1. Modified script to display the collapse arrow in job roles details to open.
2. Added modal to display confirmation on job role deletion.
3. Added and modified `hr-job-roles-controller.js` and `mainTemplate.html` to display spinner until the data is loaded completely.

### Evidence:
1. Details Arrow:
arrow

2. Job Role Delete Confirmation Dialog:
delete-dialog

3. Spinner on opening Job Roles:
spinner

4. Displaying Add Job Role Button on no Job Roles available:
displaying button

### CODE REVIEW

File: com.civicrm.hrjobroles/js/src/job-roles/controllers/modal-dialog.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        function ($scope, $modalInstance, $timeout, content, $log) {
+            $log.debug('Controller: ModalDialogCtrl');
+
+            $scope.title = content.title || 'CiviHR Job Roles';
+            $scope.msg = content.msg || '';
+            $scope.copyConfirm = content.copyConfirm || 'Yes';
+            $scope.copyCancel = content.copyCancel || 'Cancel';
+
+            $scope.confirm = function (action) {
+                $modalInstance.close(action || true);
```
Comment (sleekvick) [inline com.civicrm.hrjobroles/js/src/job-roles/controllers/modal-dialog.js]: Is this safe?

Comment (bmnepali) [inline com.civicrm.hrjobroles/js/src/job-roles/controllers/modal-dialog.js]: Any suggestion please...

Comment (bmnepali) [inline com.civicrm.hrjobroles/js/src/job-roles/controllers/modal-dialog.js]: Do you mean `$modalInstance.close(action || true);` should completely depend on action?
i.e `$modalInstance.close(action);` ?

Comment (bmnepali) [inline com.civicrm.hrjobroles/js/src/job-roles/controllers/modal-dialog.js]:
```
<div class="modal-footer">
    <button type="button" class="btn btn-default" ng-click="cancel()">{{copyCancel}}</button>
    <button type="button" class="btn btn-default" ng-click="confirm()">{{copyConfirm}}</button>
</div>
```
So, i guess this not a problem as we don't and any param to pass in `confirm() `function from template.

**Pull request merged**